code/modules/hydroponics/trays/tray.dm

```diff
@@ -101,6 +103,17 @@
 	if(closed_system)
 		flags &= ~OPENCONTAINER
 
+	var/datum/atom_hud/data/hydroponic/hydro_hud = huds[DATA_HUD_HYDROPONIC]
+	prepare_huds()
```

Comment:
Here's the problem - `prepare_huds` is way down here, **after** `RefreshParts` gets called up above. `RefreshParts` tries to update the nutrient and water overlays, but the HUD hasn't been prepared yet, so it runtimes. Considering that should prevent the HUDs from ever getting prepared, I'm surprised they're working at all...